### ISSUE TITLE
Type conflict (WebSocket)

### ISSUE THREAD

**Issue by sywka** [pcan/node-stomp-protocol#18]
### Description

Can't use library

### Steps to Reproduce

```ts
import WebSocket from "ws";
...
const wsServer = new WebSocket.Server({...});
  wsServer.on("connection", (webSocket) => {
    createStompServerSession(webSocket, {...});
  });
```

**Actual behavior:** 
I get type error on line `createStompServerSession(webSocket, {...});` for argument `webSocket`
```
Error:line (15)TS2345: Argument of type 'WebSocket' is not assignable to parameter of type 'Socket | WebSocket'.
  Type 'WebSocket' is not assignable to type 'WebSocket'. Two different types with this name exist, but they are unrelated.
```

### Versions

node: 10.5.0
typescript: 3.0.3
node-stomp-protocol: 0.3.6
ws: 6.0.0
@types/ws: 6.0.1

tsconfig.json:
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "esnext",
    "lib": ["esnext"],
    "sourceMap": true,
    "esModuleInterop": true,
    "strict": true,
    "outDir": "./dist",
    "typeRoots": [
      "node_modules/@types",
      "./typings"
    ],
    "skipLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true
  }
}
```

**Comment by pcan** [comment]
Fixed in v.0.3.7.
Thank you for reporting this.

**Comment by sywka** [comment]
Thanks for the quick fix but still there is a problem:
```
TS2345: Argument of type 'WebSocket' is not assignable to parameter of type 'Socket | WebSocket'.
  Type 'WebSocket' is not assignable to type 'import(".../node_modules/node-stomp-protocol/dist/utils").WebSocket'.
    Types of property 'on' are incompatible.
      Type '{ (event: "close", listener: (this: WebSocket, code: number, reason: string) => void): WebSocket; (event: "error", listener: (this: WebSocket, err: Error) => void): WebSocket; (event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): WebSocket; (event: "message", listener: (this: WebSocket, ...' is not assignable to type '{ (event: "message", listener: (this: WebSocket, data: WebSocketData) => void): WebSocket; (event: "error", listener: (this: WebSocket, err: Error) => void): WebSocket; (event: "close", listener: (this: WebSocket, code: number, reason: string) => void): WebSocket; }'.
        Types of parameters 'event' and 'event' are incompatible.
          Type '"message"' is not assignable to type '"close"'.
```

However now I can use type `any`:
```ts
createStompServerSession(webSocket as any, {...});
```

**Comment by pcan** [comment]
You're right. Furthermore, the `WebSocket` specification contained in `lib.es6.d.ts` is quite different and seems not compatible with `ws`. I'll open another issue tomorrow that addresses both cases.

**Comment by pcan** [comment]
@sywka Other compatibility issues have been fixed in #19. Now you should be able to use `createStompServerSession` whitout explicit cast.
